archiver.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class Archiver
+  attr_accessor :file
+
+  @modes = %i[reader writer both]
+
+  @modes.each do |mode|
+    define_method("open_file_as_#{mode}") do |file_name|
+      @file = File.open(file_name, 'r') if mode == :reader
+      @file = File.open(file_name, 'w') if mode == :writer
+      @file = File.open(file_name, 'r+') if mode == :both
+    end
+  end
+
+  def read_file(lines: false)
+    lines ? file&.read : file.readlines.map(&:chomp)
```

Comment:
Maybe the & should be in both usages of file as in both of them may be nil.

Reply:
Robocop didn't like the second & so I took it out, I didn't understood the linter violation to be honest.